src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -182,17 +182,21 @@ def _alloc_from_content(self, user_config_file=None, config=None):
             config_parser = ConfigParser()
             init_user_config_parser(config_parser)
             self.config_path = os.path.abspath(os.path.dirname(user_config_file))
-            user_config_content = config_parser.parse(
-                user_config_file,
-                pre_defined_kw_map={
-                    "<CWD>": self.config_path,
-                    "<CONFIG_PATH>": self.config_path,
-                    "<CONFIG_FILE>": os.path.basename(user_config_file),
-                    "<CONFIG_FILE_BASE>": os.path.basename(user_config_file).split(".")[
-                        0
-                    ],
-                },
-            )
+
+            try:
+                user_config_content = config_parser.parse(
+                    user_config_file,
+                    pre_defined_kw_map={
+                        "<CWD>": self.config_path,
+                        "<CONFIG_PATH>": self.config_path,
+                        "<CONFIG_FILE>": os.path.basename(user_config_file),
+                        "<CONFIG_FILE_BASE>": os.path.basename(user_config_file).split(
+                            "."
+                        )[0],
+                    },
+                )
+            except ValueError as err:
+                raise ConfigValidationError(err)
```

Comment:
This is not really what i wanted to do, rather i'd like to raise `ConfigValidationError` directly from config_parser.py:
```python
        if validate and not config_content.isValid():
            sys.stderr.write(f"Errors parsing:{config_file} \n")
            for count, error in enumerate(config_content.getErrors()):
                sys.stderr.write(f"  {count:02d}:{error}\n")
            raise ValueError(f"Parsing:{config_file} failed")

        return config_content
```
My problem is that doing so causes a circular dependency somehow. I will look further into this.

Reply:
Aaand it's done! Doomguy says so. :suspect:

Reply:
I am guessing you moved `ConfigValidationError` from `res_config.py` to `config_parser.py` which is reasonable :) . I guess alternatively we might have wanted to to have a separate file for `ConfigValidationError`, but I think this is also good.

Reply:
Should we separate the exception into a file? That would be super easy.
Also, i'll squash everything to one commit.